usr/local/share/bastille/etcupdate.sh

```diff
@@ -0,0 +1,254 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+. /usr/local/share/bastille/common.sh
+. /usr/local/etc/bastille/bastille.conf
+
+usage() {
+        # Update /etc directory while keeping user changes
+        error_notify "Usage: bastille etcupdate [option(s)(optional)] [jailname] [oldrelease] [newrelease]"
+
+        cat << EOF
+        Options:
+
+        -D | --dryrun   -- Do a dry run. Output actions to stdout but without making changes.
+        -Q | --quiet    -- Do not output actions to stdout
+
+EOF
+        exit 1
+}
+
+
+executeconditional() {
+    if [ $DRY_RUN -eq "0" ]; then
+        eval "$@"
+    fi
+}
+
+C1_C6_conditions() {
+    filelistjail=$(find "${jail_etc}" -mindepth 1 -type f)
+    for jailfile in ${filelistjail}
+    do
+        filepart=$(echo "${jailfile}" | awk -F 'etc/' '{print $NF}')
+        newbasefile="${new_basedir}/etc/${filepart}"
+        currentbasefile="${current_basedir}/etc/${filepart}"
+
+        if [ ! -f "${currentbasefile}" ]; then
+            if [ ! -f "${newbasefile}" ]; then
+                C1=$((C1+1))
+                C1ct="$C1ct${jailfile}\n"
+            else
+                C2=$((C2+1))
+                C2ct="$C2ct${jailfile}\n"
+            fi
+        fi
+
+        if [ -f "${currentbasefile}" ]; then
+            diffr=$(diff -u "${currentbasefile}" "${jailfile}")
+            if [ -z "${diffr}" ]; then
+                if [ ! -f "${newbasefile}" ]; then
+                    C3=$((C3+1))
+                    C3ct="$C3ct${jailfile}\n"
+                    cmd="rm -rf ${jailfile}"
+                    executeconditional "$cmd"
+                else
+                    C4=$((C4+1))
+                    C4ct="$C4ct${jailfile}\n"
+                    cmd="cp -p ${newbasefile} ${jailfile}"
+                    executeconditional "$cmd"
+                    # Copy keeping permissions
+                fi
+            else
```

Comment:
At this point, there have been changes to the jail's `/etc` files compared to the `currentbasefile`, so there will be a diff for C5 or C6.  Rather than have a list of diffs to revisit later (the conservative approach is to leave them alone), it would be great to have the option enter an interactive mode to approve/modify the diff here.